pkg/zkproofs/range.go

```diff
@@ -21,6 +21,19 @@ type RangeProof struct {
 	UpperBound int
 }
 
+// Caches the verifiers created for each upper bound
+type CachedRangeVerifier struct {
```

Comment:
Curious if caching belongs to cryptography/zk layer. I thought we will do the actual caching in sei-chain layer, but here just add ability to plug the verifier e.g.
```
type RangeProofVerifier struct {
	verifier *bulletproof.RangeVerifier
}
```
and then in method we just do
```
verified, err := r.verifier.Verify(proof.Proof, ciphertext.C, proofGenerators, upperBound, getTranscript())
```

Reply:
Implemented this way initially but this seems like a strictly better implementation (ie I can't see why the old method would be useful to keep around )

This is simar to what we do with the teg decryptor object